packages/account/src/Components/api-token/__tests__/api-token.spec.js

```diff
@@ -0,0 +1,326 @@
+                apiToken: jest.fn(() =>
+                    Promise.resolve({
+                        api_token: {
+                            tokens: [],
+                        },
+                    })
+                ),
+                authorized: {
+                    apiToken: jest.fn(() =>
+                        Promise.resolve({
+                            api_token: {
+                                tokens: [],
+                            },
+                        })
+                    ),
+                },
+            },
+        };
+    });
+
+    afterAll(() => {
+        document.body.removeChild(footer_portal_root_el);
+        document.body.removeChild(overlay_portal_root_el);
+    });
+
+    it('should render ApiToken component without app_settings and footer', async () => {
+        render(<ApiToken {...mock_props} />);
+
+        expect(mock_props.ws.authorized.apiToken).toHaveBeenCalled();
+
+        expect(await screen.findByText(admin_description)).toBeInTheDocument();
+        expect(await screen.findByText(payments_description)).toBeInTheDocument();
+        expect(await screen.findByText(select_scopes_msg)).toBeInTheDocument();
+        expect(await screen.findByText(token_creation_description)).toBeInTheDocument();
+        expect(await screen.findByText(token_using_description)).toBeInTheDocument();
+        expect(await screen.findByText(trade_description)).toBeInTheDocument();
+        expect(await screen.findByText(trading_info_description)).toBeInTheDocument();
+        expect(await screen.findByText(your_access_description)).toBeInTheDocument();
+        expect(await screen.findByText(view_activity_msg)).toBeInTheDocument();
+        expect(screen.queryByText(learn_more_title)).not.toBeInTheDocument();
+    });
+
+    it('should not render ApiToken component if is not mounted', async () => {
+        useIsMounted.mockImplementationOnce(() => () => false);
+
+        const { container } = render(<ApiToken {...mock_props} />);
+
+        expect(mock_props.ws.authorized.apiToken).toHaveBeenCalled();
+        expect(screen.queryByText(admin_description)).not.toBeInTheDocument();
+        expect(screen.queryByText(learn_more_title)).not.toBeInTheDocument();
+        expect(screen.queryByText(payments_description)).not.toBeInTheDocument();
+        expect(screen.queryByText(select_scopes_msg)).not.toBeInTheDocument();
+        expect(screen.queryByText(token_creation_description)).not.toBeInTheDocument();
+        expect(screen.queryByText(token_using_description)).not.toBeInTheDocument();
+        expect(screen.queryByText(trade_description)).not.toBeInTheDocument();
+        expect(screen.queryByText(trading_info_description)).not.toBeInTheDocument();
+        expect(screen.queryByText(your_access_description)).not.toBeInTheDocument();
+        expect(screen.queryByText(view_activity_msg)).not.toBeInTheDocument();
+
+        const loader = container.querySelector('.initial-loader');
```

Comment:
instead of testing with classname, i think it is better to mock loader component and test
```
jest.mock('@deriv/components', () => ({
    ...jest.requireActual('@deriv/components'),
    Loading: () => <div>Loading</div>,
}));
```

Reply:
it's a great note, thanks @amina-deriv . Why didn't I think about it... :)